util.sh

```diff
@@ -149,3 +149,34 @@ function exit_on_common {
             ;;
     esac
 }
+
+# ============================================================================
+# SELF UPDATE
+# ============================================================================
+
+util_url=${util_url:-}
+util_check_min=60
+SOURCE=${BASH_SOURCE:-}
+
+# if SOURCE is NOT symlink (local testing)
+# and url is set
+# and url points to non-pinned main branch
+# and file is older than $util_check_min
+if [ ! -L ${SOURCE} ] \
+    && [ -n "${util_url}" ] \
+    && [[ "$util_url" == *"/main/"* ]] \
+    && test $(find ${SOURCE} -mmin +${util_check_min}); then
+
+    # only then check if its up to-date
+    if ! cat ${SOURCE} | sha256sum --check --quiet <(curl -s $util_url | sha256sum) &> /dev/null; then
```

Comment:
this should address the my file locally is out-of-date

note it does not take any action implicitly. it simply notifies that `util.sh` is out-of-date but its up to the user to decide whether to actually do anything.

also it checks for things like whether url is set, so that when we decide to embed it within other repos to make them open-source, this will be no-op